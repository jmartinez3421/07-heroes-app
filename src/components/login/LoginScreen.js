import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ({history}) => {
    const {dispatch} = useContext(AuthContext);
    const handleLogin = () => {
        
        const lastPath = localStorage.getItem('lastPath') || '/';
        // history.push('/');
        history.replace(lastPath); //Lo hacemos con replace para que no exista el login para volver atr√°s
        
        dispatch({
            type: types.login,
            payload: {
                name: 'Jordi'
            }
        })

        
    }

    return (
        <div className='container mt-5'>
            <h1 id='login'>Login</h1>
            <hr />

            <button 
                className='btn btn-outline-info'
                onClick={handleLogin}
            >
                Login
            </button>
        </div>
    )
}
