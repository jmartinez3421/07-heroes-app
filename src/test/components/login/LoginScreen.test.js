import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from "react-router";

import { AuthContext } from "../../../auth/AuthContext";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { types } from '../../../types/types';

describe('Pruebas sobre el componente <LoginScreen />', () => {

    const history = {
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }
    
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <LoginScreen history={history}/>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('button').exists()).toBe(true);
    });

    test('Debe de realizar el dispatch y la navegacion', () => {
        const handleClick = wrapper.find('button').prop('onClick');

        handleClick();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'Jordi'
            }
        });
        expect(history.replace).toHaveBeenCalled();

        localStorage.setItem('lastPath', '/search');
        handleClick();
        expect(history.replace).toHaveBeenCalledWith('/search')
    });
    
    
})
