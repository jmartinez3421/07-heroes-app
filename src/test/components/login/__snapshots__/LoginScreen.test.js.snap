// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <LoginScreen /> Debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1
    id="login"
  >
    Login
  </h1>
  <hr />
  <button
    className="btn btn-outline-info"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
